#
# This patch adds a missing include to avoid compilation error.
#
# Recently, OpenSSH includes were refactored, so that header files no longer
# include system header files. System header files are now included in
# sources only.
#
# kex.h references sig_atomic_t, but no longer includes signal.h.
# Now every file including kex.h must include signal.h. gss-genr.c failed
# to do so, which resulted in unknown type compilation error.
#
# The patch has been accepted by upstream and will be part of 6.9 release.
#    https://bugzilla.mindrot.org/show_bug.cgi?id=2402
#
# When upgrading to some release >=6.9, this patch will be dropped.
#
--- a/gss-genr.c	
+++ a/gss-genr.c	
@@ -34,6 +34,7 @@ 
 #include <limits.h>
 #include <stdarg.h>
 #include <string.h>
+#include <signal.h>
 #include <unistd.h>
 
 #include "xmalloc.h"

